docs/troubleshooting_guide.md

```diff
@@ -0,0 +1,358 @@
+    "cluster_name": "dec-test-cluster",
+    "cluster_domain": "dec-test-home",
+    "public_net": "dec-test-public_interface",
+    "private_net": "dec-test-private",
+    "scan_ip1": "172.16.30.20",
+    "scan_ip2": "172.16.30.21",
+    "scan_ip3": "172.16.30.22",
+    "dg_name": "DATA",<======================================== this parameter was missing in this file's this line
+    "nodes": [
+      {  "node_name": "racnode1.orcl",
+         "host_ip": "172.16.30.1",
+         "vip_name": "racnode1-vip.orcl",
+         "vip_ip": "172.16.30.11"
+      },
+      {  "node_name": "racnode2.orcl",
+         "host_ip": "172.16.30.2",
+         "vip_name": "racnode2-vip.orcl",
+         "vip_ip": "172.16.30.12"
+      }
+    ]
+  }
+]
+
+```
+
+#### 2. Error manifestation with respect to an invalid disk group name:
+* For following error:
+
+```bash
+TASK [rac-gi-setup : rac-gi-install | Get symlinks for devices] ************************************************************************************************
+task path: /home/janedoe/git-clone-holder/bms-toolkit/roles/rac-gi-setup/tasks/rac-gi-install.yml:127
+File lookup using /home/janedoe/git-clone-holder/bms-toolkit/asm_disk_config.json as file
+fatal: [racnode1.orcl]: FAILED! => {
+    "msg": "Invalid data passed to 'loop', it requires a list, got this instead: . Hint: If you passed a list/dict of just one element, try adding wantlist=True to your lookup invocation or use q/query instead of lookup."
+}
+```
+
+* Check that: `dg_name=DATA` points to a valid diskgroup name.
+* Rationale: the expression:
+```bash
+"{{ asm_disks | json_query('[?diskgroup==`' + hostvars[groups['dbasm'].0]['dg_name'] + '`].disks[*].blk_device') | list | join() }}"
+```
+
+... translates to something similar to the following:
+```bash
+"{{ asm_disks | json_query('[?diskgroup==`'DATA'`].disks[*].blk_device') | list | join() }}"
+```
+
+where the 'DATA' is the incorrect DG name supplied in cluster_config.json file's `"dg_name": "DATA"`
+
+#### 3. Error manifestation due to an invalid network interface name
+* Error:
+```bash
+TASK [rac-gi-setup : rac-gi-install | Create GI response file] *************************************************************************************************
+fatal: [racnode1.orcl]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: 'dict object' has no attribute u'dec-test-public_interface'"}
+```
+
+* Fix:
+Ensure that the entries for the n/w interfaces are correctly defined:
+```bash
+✔ ~/git-clone-holder/bms-toolkit [master L|✚ 114…15] 
+20:02 $ cat cluster_config.json
+20:03 $ cat cluster_config.json.enough_fun
+[
+  {
+    "scan_name": "dec-test-scan-cluster",
+    "scan_port": "1521",
+    "cluster_name": "dec-test-cluster",
+    "cluster_domain": "dec-test-home",
+    "public_net": "dec-test-public_interface",   <==================== this and 
+    "private_net": "dec-test-private",           <==================== this should reflect the actual interfaces on the BM hosts 
```

Comment:
Might not necessarily be a BM host;  should reflect an active interface on the target host.  Maybe even include syntax on how to figure this out?

Reply:
Edited to remove BM.
Also, the following existing sections throw more light on public and private network details.